engine.py

```diff
@@ -891,3 +930,128 @@ def _get_dialog_parent(self):
+                    )
+                    break
+            self.__node_handlers[node_category][node_type_name] = hook_instance
+                    
+        return hook_instance
+    
+    def all_sgtk_nodes(self):
+        """
+        Get all teh nodes in the scene that contains sgtk parameters.
+
+        :rtype: list(:class:`hou.Node`)
+        """
+        node_type_categories = hou.nodeTypeCategories()
+        node_handler_settings = self.get_setting("node_handlers")
+
+        def key_func(item):
+            return item["node_category"]
+
+        grouped = itertools.groupby(
+            sorted(node_handler_settings, key=key_func),
+            key=key_func
+        )
+
+        for category_name, node_handlers in grouped:
+            category = node_type_categories[category_name]
+            for node_handler in node_handlers:
+                node_type_name = node_handler["node_type"]
+                node_type = hou.nodeType(category, node_type_name)
+                for node in node_type.instances():
+                    parm = node.parm("sgtk_identifier")
+                    if parm:
+                        self.log_debug(node.path())
+                        yield node
+
+    def remove_sgtk_parms(self, node):
+        """
+        Remove all sgtk parms on the given node.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        handler = self.node_handler(node)
+        handler.remove_sgtk_parms(node)
+
+    def remove_all_sgtk_parms(self):
+        """
+        Remove all sgtk parms in the scene.
+        """
+        for node in self.all_sgtk_nodes():
+            self.remove_sgtk_parms(node)
+
+    def restore_sgtk_parms(self, node):
+        """
+        Restore all sgtk parms on the given node.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        handler = self.node_handler(node)
+        handler.restore_sgtk_parms(node)
+
+    def restore_all_sgtk_parms(self):
+        """
+        Restore all sgtk parms in the scene.
+        """
+        for node in self.all_sgtk_nodes():
+            self.restore_sgtk_parms(node)
+
+def _refresh_variables_and_node_handlers_callback(event):
+    """
+    Callback to refresh all variables and node handler classes.
+
+    This can happen on load, save and clear.
+
+    :param event: A :class:`hou.hipFileEventType` event.
+    """
+    valid_event_types = (
+        hou.hipFileEventType.AfterSave,
+        hou.hipFileEventType.AfterLoad,
+        hou.hipFileEventType.AfterClear
+    )
+    if event not in valid_event_types:
+        return
+
+    engine = sgtk.platform.current_engine()
+    if not engine:
+        return
+
+    engine.reset_node_handlers()  # reset node handlers as they may change between contexts
+    
+    engine.update_variables()
+
+    for node in engine.all_sgtk_nodes():
+        handler = engine.node_handler(node)
+        if node.parm("sgtk_folder"):
+            handler._refresh_file_path(node)
```

Comment:
Simplify to just `if engine`

```suggestion
    if engine:
        # reset node handlers as they may change between contexts
        engine.reset_node_handlers()
        engine.update_variables()
        for node in engine.all_sgtk_nodes():
            handler = engine.node_handler(node)
            if node.parm("sgtk_folder"):
                handler._refresh_file_path(node)
```